src/client.js

```diff
@@ -134,6 +107,34 @@ class Client extends events.EventEmitter {
     }
   }
 
+  _startConnection() {
+    const newConn = new Connection(this._retirementDelay);
+
+    // don't actually start connecting until at least minReconnectionInterval has passed
+    const earliestConnectionTime = this.lastConnectionTime + this._minReconnectionInterval;
+    const now = new Date().getTime();
+    setTimeout(async () => {
+      if (newConn.state !== 'waiting') {
```

Comment:
If `null` is returned I wonder if `newConn.once` can throw error ?

Reply:
Good question. The return from this setTimeout function isn't important -- and anyway the it won't be called until after the timeout, so `newConn.once` has already been called by that time.